client/Dockerfile

```diff
@@ -0,0 +1,19 @@
+FROM node:18 AS build
+
+WORKDIR /app
+
+COPY package.json yarn.lock ./
+RUN yarn install --frozen-lockfile
+COPY . ./
+
+ARG DEMO
+ENV VITE_DEMO ${DEMO}
+ARG CAPTCHA_KEY
```

Comment:
Those variables should not be set in docker build args but by docker environment variables. If those environment specific variables are set in the docker image itself, the image will not longer be portable to multiple environments without rebuilding.